wip-mariocao-min-difficulty-dr.md

```diff
@@ -0,0 +1,213 @@
+
+Data request eligibility is computed by using the Reputation-based Proof of Eligibility (_[RepPoE]_) algorithm. It determines if an identity is eligible or not based on reputation metrics and the data request replication factor.
+
+Due to the stochastic nature of the algorithm and because not all eligible committee members might be available for a given data request, the _RepPoE_ algorithm includes a backup factor to increase the odds of having enough candidate witnesses for constituting the committee. Therefore, depending on how this algorithm is fine-tuned, the number of eligible identities will vary, and, consequently, the number of commit transactions broadcasted to the network.
+
+Analogously to the [WIP-0012], it becomes clear that an upper bound must be enforced on data request eligibility, which translates to enforcing a lower bound on data request mining difficulty. This will prevent potential floodings of `commit` which may lead to an accidental [denial-of-service attack][DoS].
+
+
+### Current implementation of Reputation-based Proof of Eligibility (_RepPoE_)
+
+A simplified version of the reputation-based proof of eligibility algorithm (a.k.a. _RepPoE_) could be defined as:
+
+```
+P_i = reputation_i / R * rf
+```
+
+where:
+
+- `P_i` is the probability of an identity being eligible
+- `reputation_i` is the reputation
+- `R` is the total active reputation in the system
+- `rf` is the data request replication factor
+
+The probability `P` will be used as a static threshold that sets the maximum value until which a _RepPoE_ proof can be considered valid. This threshold will be compared with the output of a [VRF] of the identity which takes as seed the data request hash and the last checkpoint beacon of the chain (i.e. epoch and block hash).
+
+Later, a factor `alpha` was introduced to adjust the data request mining eligibilities by taking into consideration the state of the reputation system. This factor aims to compensate for the over-representation of identities with significant reputation scores and thus affecting the average amount of eligible identities per data request.
+
+```
+P_i = reputation_i / R * rf * alpha
+```
+
+where:
+
+- `alpha` is a factor as defined in [Dynamic Threshold]
+
+The last modification to the _RepPoE_ algorithm was the usage of a metric labeled as _identity eligibility_ instead of the reputation score as the numerator of the previous formula:
+
+```
+P_i = eligibility_i / R * rf * alpha
+```
+
+where:
+
+- `eligibility_i` is the computed identity eligibility as mentioned in [witnet-rust#1496] and later described
+
+The _eligibility_ metric aims to achieve a "more linear" distribution of data request mining probabilities across identities. As shown in the images below, the eligibility can be understood as a redistribution of reputation across identities following a _right trapezoid_ shape.
```

Comment:
> What's the subtext behind the quotation marks?

First, it was to remind me to review that adjective.
I wanted to avoid using "fairer distribution", which can be considered a subjective qualifier.

Anyhow, I forgot to remove the double quotes.